packages/core/src/modal/modal.stories.ts

```diff
@@ -23,118 +24,265 @@ export default {
+    myOverlay.closeOverlay();
+  }
+
   return html`
-    <cds-demo popover>
-      <cds-modal size="lg" __demo-mode>
-        <cds-modal-header>
-          <h3 cds-text="title">My Modal</h3>
-        </cds-modal-header>
-        <cds-modal-content>
-          <p cds-text="body">Lorem Ipsum</p>
-        </cds-modal-content>
-        <cds-modal-actions>
-          <cds-button action="outline">Cancel</cds-button>
-          <cds-button>Ok</cds-button>
-        </cds-modal-actions>
-      </cds-modal>
-    </cds-demo>
+    <cds-button status="primary" type="button" @click=${showLargeModal}>Show Large Modal</cds-button>
+    <cds-modal hidden size="lg" id="${modalId}">
+      <cds-modal-header>
+        <h3 cds-text="title"></h3>Large Modal</h3>
+      </cds-modal-header>
+      <cds-modal-content>
+        <p cds-text="body">Place holder text for the large modal example.</p>
+        <p cds-text="body">Lorem Ipsum</p>
+      </cds-modal-content>
+      <cds-modal-actions>
+        <cds-button @click=${hideLargeModal} action="outline">Cancel</cds-button>
+        <cds-button @click=${hideLargeModal}>Ok</cds-button>
+      </cds-modal-actions>
+    </cds-modal>
   `;
 }
 
 /** @website */
 export function extraLarge() {
+  let initted = false;
+  const modalId = 'modal-xlarge';
+
+  function showXLargeModal() {
+    const myOverlay = document.getElementById(modalId) as CdsModal;
+    myOverlay.removeAttribute('hidden');
+
+    if (!initted) {
+      myOverlay.addEventListener('closeChange', () => {
+        myOverlay.setAttribute('hidden', 'true');
+      });
+      initted = true;
+    }
+  }
+
+  function hideXLargeModal() {
+    const myOverlay = document.getElementById(modalId) as CdsModal;
+    myOverlay.closeOverlay();
+  }
+
   return html`
-    <cds-demo popover>
-      <cds-modal size="xl" __demo-mode>
-        <cds-modal-header>
-          <h3 cds-text="title">My Modal</h3>
-        </cds-modal-header>
-        <cds-modal-content>
-          <p cds-text="body">Lorem Ipsum</p>
-        </cds-modal-content>
-        <cds-modal-actions>
-          <cds-button action="outline">Cancel</cds-button>
-          <cds-button>Ok</cds-button>
-        </cds-modal-actions>
-      </cds-modal>
-    </cds-demo>
+    <cds-button status="primary" type="button" @click=${showXLargeModal}>Show Extra Large Modal</cds-button>
+    <cds-modal hidden size="xl" id="${modalId}">
+      <cds-modal-header>
+        <h3 cds-text="title">Large Modal</h3>
+      </cds-modal-header>
+      <cds-modal-content>
+        <p cds-text="body">Place holder text for the extra large modal example.</p>
+        <p cds-text="body">Lorem Ipsum</p>
+      </cds-modal-content>
+      <cds-modal-actions>
+        <cds-button @click=${hideXLargeModal} action="outline">Cancel</cds-button>
+        <cds-button @click=${hideXLargeModal}>Ok</cds-button>
+      </cds-modal-actions>
+    </cds-modal>
   `;
 }
 
 /** @website */
 export function focusTrap() {
```

Comment:
Looking at the interactive demos this one seems to have a layout issue for the demo https://5fda9d2e0ea1383e0d1a73c2--vmware-clarity.netlify.app/storybook/core/?path=/story/components-modal--page#managing-focus-in-a-modal